trappy/ftrace.py

```diff
@@ -168,28 +168,21 @@ class definitions list.  Otherwise, register a class to parse
     def __populate_data(self, fin, cls_for_unique_word, window, abs_window):
         """Append to trace data from a txt trace"""
 
-        def contains_unique_word(line, unique_words=cls_for_unique_word.keys()):
-            for unique_word in unique_words:
-                if unique_word in line:
-                    return True
-            return False
-
         actual_trace = itertools.dropwhile(self.trace_hasnt_started(), fin)
         actual_trace = itertools.takewhile(self.trace_hasnt_finished(),
                                            actual_trace)
 
         for line in actual_trace:
-            if not contains_unique_word(line):
-                self.lines += 1
-                continue
+            trace_class = None
             for unique_word, cls in cls_for_unique_word.iteritems():
                 if unique_word in line:
                     trace_class = cls
                     if not cls.fallback:
                         break
             else:
                 if not trace_class:
-                    raise FTraceParseError("No unique word in '{}'".format(line))
+                    self.lines += 1
+                    continue
```

Comment:
This seems to be again changing the behaviour of the code. We have a unique word but no class for that.